commons/hu.qgears.opengl.commons/kms/drmScreenshot.cpp

```diff
@@ -101,67 +101,66 @@ int main (int argc, char ** argv)
 	}
 
 	/* find the first available connector with modes */
-	for(i=0; i < resources->count_connectors; ++i){
+	for(i=0; i < resources->count_connectors; ++i) {
 		connector = drmModeGetConnector(fd, resources->connectors[i]);
 		if(connector != NULL){
-			fprintf(stderr, "connector %d found\n", connector->connector_id);
-			if(connector->connection == DRM_MODE_CONNECTED
-				&& connector->count_modes > 0)
-				break;
-			drmModeFreeConnector(connector);
-		}
-		else
+			fprintf(stderr, "connector %d found; mode count: %d; state: ", 
+			    connector->connector_id, connector->count_modes);
+			    
+			if(connector->connection == DRM_MODE_CONNECTED) {
+				fprintf(stderr, "connected\n");
+				
+				if (first_good_connector == NULL) {
+				    first_good_connector = connector;
+				}
+			} else {
+			    fprintf(stderr, "disconnected\n");
+			}
+//				break;
+//			drmModeFreeConnector(connector);
```

Comment:
1. A régi kód felszabadította ( drmModeFreeConnector ) azokat a konnektorokat, amik nem kerültek kiválasztásra. Ez a mostani változatból hiányzik.
2. A first_good_connector megtalálása után megszakíthatnánk a ciklust, nem feltétlenül muszáj végigmenni az összesen, nem ?
3. Az eredeti kód akkor választotta ki a connectort ha (connector->connection == DRM_MODE_CONNECTED && connector->count_modes > 0 ). A connector->count_modes > 0 rész hiányzik az újból. Vagy ez a javítás lényege?

Reply:
1. Jogos, visszaállítom a felszabadítást.
2. Debug-céllal maradt így, hogy lehessen látni, hogy egyáltalán mit ismer fel a keretrendszer. Ha alaposak akarnánk lenni, még kellene a kódba egyértelmű konnektor-azonosítás is, pl. hogy melyik az LVDS, melyik a HDMI, ezen kívül kellene programindítási paraméter, hogy ezek közül melyikről kell képernyőképet készíteni, legalábbis, ha szem előtt tartjuk az esetlegesen alkalmazó projektekben egyre inkább teret nyerő, explicit konfigurációt (azaz a nem feltétlenül determinisztikus, plug-and-play jellegű, alapértelmezett működésektől való elmozdulást), mint irányelvet. Jelen formájában a kód feltételezi, hogy az elsőként megtalált monitorról készítünk képet, és mivel az eszközünkhöz mindig 1 monitor csatlakozik, ez így ugyanúgy jó, mint az eredeti változat. Ha az lesz a döntés, hogy behatóbban kell a kmsgl-lel foglalkozni, ezt végig kell majd vinni.
3. Igazából mindegy, mert ha "connected" állapotú egy konnektor, de nincs értelmezhető üzemmódja, akkor a keretrendszer a hibás, a mi követelményeink alapján. A gyakorlatban talán akkor lehet 0 az üzemmódok száma, ha alszik az adott monitor, de ilyen sem fordulhat elő a jelenlegi követelményrendszer alapján, tehát ez is körbe kellene járni, ha alaposak akarnánk lenni. De végül is ártani nem árt az a feltétel oda, visszateszem.